src/App.tsx

```diff
@@ -26,70 +27,123 @@ type ReorderOptions = {
+  visibleGoods.sort((goodA, goodB) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return goodA.localeCompare(goodB);
+
+      case SortType.LENGTH:
+        return goodA.length - goodB.length;
+
+      case SortType.NONE:
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  handleNameSort = () => {
+    this.setState({ sortType: SortType.ALPHABET });
+  };
+
+  handleLengthSort = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  handleReverseSort = () => {
+    this.setState(state => ({ isReversed: !state.isReversed }));
+  };
+
+  handleReset = () => {
+    this.setState({
+      isReversed: false,
+      sortType: SortType.NONE,
+    });
+  };
+
+  render() {
+    const goods = getReorderedGoods(goodsFromServer, this.state);
+
+    const notChanged = this.state.sortType === SortType.NONE
+      && !this.state.isReversed;
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            onClick={this.handleNameSort}
+            className={classNames('button is-info', {
+              'is-light': this.state.sortType !== SortType.ALPHABET,
+            })}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            onClick={this.handleLengthSort}
+            className={classNames('button is-success', {
+              'is-light': this.state.sortType !== SortType.LENGTH,
+            })}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            onClick={this.handleReverseSort}
+            className={classNames('button is-warning', {
+              'is-light': !this.state.isReversed,
+            })}
+          >
+            Reverse
+          </button>
+
+          {!notChanged && (
```

Comment:
actually it is quite difficult to process in my head, I read it as `not not changed`. Better to use positive condition here. Let's make the variable like `areGoodsChanged` or `areGoodsOriginal`